changes/9709.outlier_detection.rst

```diff
@@ -0,0 +1,2 @@
+Copy input data before processing to ensure input datamodels are not modified.
```

Comment:
might be good to specify here that the high-memory-usage large mosaic cases were not modified here, although I don't know exactly how to state it.